src/SelfServe/SqlX/SqlX.rp.ts

```diff
@@ -128,3 +128,105 @@ export const refreshDedicatedGatewayProvisioning = async (): Promise<RefreshResu
+  return `/subscriptions/${subscriptionId}/resourceGroups/${resourceGroup}/providers/Microsoft.DocumentDB/databaseAccounts/${name}`;
+};
+
+// Query ARM to get regions
+export const getReadRegions = async (): Promise<Array<string>> => {
+  try {
+    const readRegions = new Array<string>();
+
+    // Query ARM
+    const response = await armRequestWithoutPolling<ReadRegionsResponse>({
+      host: configContext.ARM_ENDPOINT,
+      path: getGeneralPath(userContext.subscriptionId, userContext.resourceGroup, userContext.databaseAccount.name),
+      method: "GET",
+      apiVersion: "2021-04-01-preview",
+    });
+
+    // Parse ARM Response to determine read regions
+    if (response.result.location !== undefined) {
+      readRegions.push(response.result.location.replace(" ", "").toLowerCase());
+    } else {
+      for (let i = 0; i < response.result.locations.length; i++) {
+        readRegions.push(response.result.locations[i].locationName.replace(" ", "").toLowerCase());
+      }
+    }
+    return readRegions;
+  } catch (err) {
+    alert(err);
+    // TODO: Log some failure
+    return new Array<string>();
+  }
+};
+
+// Get Fetch Prices Path for Dedicated Gateway Product
+const getFetchPricesPathForRegion = (subscriptionId: string): string => {
+  return `/subscriptions/${subscriptionId}/providers/Microsoft.CostManagement/fetchPrices`;
+};
+
+// Data Model for Fetch Prices API
+interface FetchPricesResponse {
+  Items: Array<PriceItem>;
+  NextPageLink: string | undefined;
+  Count: number;
+}
+
+interface PriceItem {
+  retailPrice: number;
+  skuName: string;
+}
+
+// Query ARM Fetch Prices to get pricing
+export const getPriceMap = async (readRegions: Array<string>): Promise<Map<string, Map<string, number>>> => {
+  try {
+    // Initialize empty PriceMap
+    const priceMap = new Map<string, Map<string, number>>();
+
+    // Query ARM
+    for (let i = 0; i < readRegions.length; i++) {
+      const regionPriceMap = new Map<string, number>();
+
+      // Make actual query
+      const response = await armRequestWithoutPolling<FetchPricesResponse>({
+        host: configContext.ARM_ENDPOINT,
+        path: getFetchPricesPathForRegion(userContext.subscriptionId),
+        method: "POST",
+        apiVersion: "2020-01-01-preview",
+        queryParams: {
+          filter:
+            "armRegionName eq " +
+            readRegions[i] +
+            " and serviceFamily eq 'Databases' and productName eq 'Azure Cosmos DB Dedicated Gateway - General Purpose'",
+        },
+      });
+
+      // Parse response to create price mapping for current region
+      for (let j = 0; j < response.result.Items.length; j++) {
+        regionPriceMap.set(response.result.Items[j].skuName, response.result.Items[j].retailPrice);
+      }
+
+      // Insert region's PriceMap to overall PriceMap
+      priceMap.set(readRegions[i], regionPriceMap);
+    }
+
+    return priceMap;
+  } catch (err) {
+    alert(err);
```

Comment:
remove alert. refer getDedicatedGatewayResource for throwing on error after logging telemetry.